ReSharper.FSharp/src/FSharp.Psi.Features/src/Intentions/FunctionAnnotationAction.fs

```diff
@@ -77,60 +51,56 @@ type FunctionAnnotationAction(dataProvider: FSharpContextActionDataProvider) =
-        let treeParameters = parameterOwner.Parameters |> Seq.toList
+        let typedTreeParameters =
+            parameterOwner.Parameters
+            |> Seq.zip fSharpFunction.CurriedParameterGroups
+            |> Seq.toList
             
         // Annotate function parameters
         let factory = namedPat.CreateElementFactory()
-        let childrenToModify =
-            treeParameters
-            |> Seq.choose(
-                function
-                | :? ILocalReferencePat as ref ->
-                    Some {ParameterNameNode = ref; ParameterNodeInSignature = ref}
+        for fSharpTypes, parameter in typedTreeParameters do
+            let paramName =
+                match parameter with
+                | :? ILocalReferencePat as ref -> Some ref.SourceName
                 | :? IParenPat as ref ->
-                    let parameterNameNode =
-                        ref.Children()
-                        |> Seq.choose(function | :? ILocalReferencePat as pat -> Some pat | _ -> None)
-                        |> Seq.tryExactlyOne
-                    parameterNameNode
-                    |> Option.map(fun namedNode ->
-                        {ParameterNameNode = namedNode;
-                          ParameterNodeInSignature = ref})
-                | _ -> None)
-            |> Seq.toList
-            
-        // Parse FSharp file, in order to get tooltips
-        let parsedFile =
-            namedPat.FSharpFile.GetParseAndCheckResults(true, opName)
-            |> Option.defaultWith(fun () -> failwith "Unable to parse FSharp file")
-        let checkResults = parsedFile.CheckResults
-        let document = namedPat.FSharpFile.GetSourceFile().Document
-            
-        // Replace parameters with their tooltip values - each parameter is associated with a list of FSharpType instances
-        // and so `symbolUse.DisplayContext.WithShortTypeNames`, as used with return type can't be used.
-        childrenToModify
-        |> List.map(fun ref ->
-            getParameterTooltip document checkResults ref
-            |> Option.map (fun signature ->
-                let typedPat = factory.CreateTypedPatInParens(signature)
-                fun () -> PsiModificationUtil.replaceWithCopy ref.ParameterNodeInSignature typedPat)
-            |> Option.defaultValue(fun () -> ()))
-        |> List.iter(fun x -> x()) // Only modify the document once the tooltips have been determined
+                    ref.Pattern.As<IReferencePat>()
+                    |> Option.ofObj
+                    |> Option.map(fun x -> x.SourceName)
+                | _ -> None
+            match paramName with
+            | Some name ->
+                // If the parameter is not curried, there will be multiple types pertaining to it
+                let subTypeUsages =
+                    fSharpTypes
+                    |> Seq.map(fun fSharpType ->
+                        fsharpSymbolUse.DisplayContext.WithShortTypeNames(true)
```

Comment:
`WithShortTypeNames(true)` only strips namespaces/modules that are no imported, and this is considered a bit easier to grasp in tooltips/completion than reading the full names.

Unfortunately, we should not do it here, since it'll make the added types unresolved, like in this example:
```fsharp
let f x =
    x |> dict
```